kotlin-styled/src/main/kotlin/styled/StyledComponents.kt

```diff
@@ -81,29 +81,70 @@ inline fun <P : CustomStyledProps> RElementBuilder<P>.css(noinline handler: Rule
 /**
  * @deprecated Use [keyframes] and [css] instead
  */
-@Suppress("NOTHING_TO_INLINE")
-inline fun keyframesName(string: String): String {
-    val keyframe = keyframes(string)
-    injectGlobal(css(keyframe))
+fun keyframesName(string: String): String {
+    val keyframes = keyframes(string)
+    val keyframesInternal = css(keyframes.rules).asDynamic()
+    val name = keyframes.getName()
+    if (keyframesInternal is String) {
+        injectGlobals(arrayOf(
+            "@-webkit-keyframes $name {$keyframesInternal}",
+            "@keyframes $name {$keyframesInternal}"
+        ))
+    } else {
+        injectGlobals(keyframesInternal)
+    }
+
+    return keyframes.getName()
+}
+
+private fun injectGlobals(strings: Array<String>) {
+    val globalStyle = createGlobalStyle(strings)
+    Promise.resolve(Unit).then {
+        GlobalStyles.add(globalStyle)
+    }
+}
 
-    return keyframe.getName()
+private external interface GlobalStylesComponentProps: RProps {
+    var globalStyles: List<Any>
 }
 
-private var globalStylesCounter = 0
+private object GlobalStyles {
+    private val component = functionalComponent<GlobalStylesComponentProps> { props ->
+        props.globalStyles.forEach {
+            child(it, jsObject {}, emptyList())
+        }
+    }
+
+    private val root by kotlin.lazy {
+        val element = window.document.body!!.appendChild(window.document.createElement("div")) as Element
```

Comment:
```suggestion
        val element = document.body!!.appendChild(document.createElement("div")) as Element
```

Reply:
Check showed that `document` can't be resolved